EmptySet with S

dnabanita7 opened the pull request:
Since,Singleton classes have instances that are the same name as the class.So,
EmptySet should return its instance rather than its class name.
As well,I changed the test_args.py to test ``if S.EmptySet is EmptySet()``.
Fixes #16337

<!-- BEGIN RELEASE NOTES -->
* sets
     * changed __init__.py to show instance of EmptySet when being called rather than showing its classname
<!-- END RELEASE NOTES -->

dnabanita7 committed:
EmptySet with S

Since,Singleton classes have instances that are the same name as the class.So,
EmptySet should return its instance rather than its class name.
As well,I changed the test_args.py to test ``if S.EmptySet is EmptySet()``.
Fixes #16337

dnabanita7 committed:
No changes

oscarbenjamin commented on sympy/concrete/summations.py:
```diff
@@ -946,9 +946,15 @@ def eval_sum(f, limits):
+    #if not(f.is_convergent) and expr.subs(1,a+i).is_positive:
+    #    return oo
+    # elif not(f.is_convergent) and expr.subs(i,a+i).is_negative:
+    #    return -oo
```
Are these lines left over from debugging?

Naba7 replied:
Yes,but this is from other PR.Then,I didn't know how to create branches.So,here is the problem.I will clear this thing over here

oscarbenjamin commented on sympy/concrete/summations.py:
```diff
@@ -946,9 +946,15 @@ def eval_sum(f, limits):
     # Try to do it symbolically. Even when the number of terms is known,
     # this can save time when b-a is big.
     # We should try to transform to partial fractions
+    #if b != oo or a == -oo:
```
And this one?

Naba7 replied:
same mistake

oscarbenjamin commented:
I'm not sure I understand what the changes here are doing. Can you give clear examples of input and output before and after this change?

Naba7 commented:
I have just changed the __init__.py .Plus I am changing importing of EmptySet in different functions.I am making a pr  for that.

dnabanita7 committed:
changed importing of EmptySet

Although,I have changed the import error of EmptySet,I still get the error that EmptySet can't be
dispatched as it is a non-type object as it is getting its signature from S.EmptySet not from
sympy.sets.sets

Naba7 commented:
Although,I have changed the import error of EmptySet,I still get the error that EmptySet can't be
dispatched as it is a non-type object as it is getting its signature from S.EmptySet not from
sympy.sets.sets

asmeurer commented:
Travis might not be running because of the merge conflicts. I would suggest fixing them (you will need to do it anyway).
